packages/core/src/odata/v4/filter/filter-lambda-expression.ts

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { FieldType } from '../../common/selectable';
+import {
+  Filter,
+  Filterable,
+  FilterLink,
+  FilterList,
+  isFilter
+} from '../../common/filter';
+import { EntityBase } from '../../common';
+
+/**
+ * @experimental This is experimental and is subject to change. Use with caution.
+ */
+export type FilterLambdaOperator = 'any' | 'all';
+
+/**
+ * @experimental This is experimental and is subject to change. Use with caution.
+ */
+export class FilterLambdaExpression<
+  EntityT extends EntityBase,
+  FieldT extends FieldType
+> {
+  constructor(
+    public filters: (
+      | Filter<EntityT, FieldT>
+      | FilterLink<EntityT>
+      | FilterList<EntityT>
+    )[],
+    public lambdaOperator: FilterLambdaOperator
+  ) {}
```

Comment:
I think we can save ourselves some trouble by always making the filters a filterlist. You can also consider moving that down to the `any` and `all` functions, I think both approaches are fine.

```suggestion
  public filter: FilterList<EntityT>;
  constructor(
   filters: (
      | Filter<EntityT, FieldT>
      | FilterLink<EntityT>
      | FilterList<EntityT>
    )[],
    public lambdaOperator: FilterLambdaOperator
  ) {
    this.filter = and(...filters);
  }
```